src/ResourceManagement/SqlManagement/Microsoft.Azure.Management.Sql/Generated/Models/RecommendedIndex.cs

```diff
@@ -0,0 +1,175 @@
+        /// <param name="indexType">The type of index (CLUSTERED,
+        /// NONCLUSTERED, COLUMNSTORE, CLUSTERED COLUMNSTORE). Possible
+        /// values include: 'CLUSTERED', 'NONCLUSTERED', 'COLUMNSTORE',
+        /// 'CLUSTERED COLUMNSTORE'</param>
+        /// <param name="schema">The schema where table to build index over
+        /// resides</param>
+        /// <param name="table">The table on which to build index.</param>
+        /// <param name="columns">Columns over which to build index</param>
+        /// <param name="includedColumns">The list of column names to be
+        /// included in the index</param>
+        /// <param name="indexScript">The full build index script</param>
+        /// <param name="estimatedImpact">The estimated impact of doing
+        /// recommended index action.</param>
+        /// <param name="reportedImpact">The values reported after index
+        /// action is complete.</param>
+        public RecommendedIndex(string location, string name = default(string), string id = default(string), string type = default(string), System.Collections.Generic.IDictionary<string, string> tags = default(System.Collections.Generic.IDictionary<string, string>), RecommendedIndexActions? action = default(RecommendedIndexActions?), RecommendedIndexStates? state = default(RecommendedIndexStates?), System.DateTime? created = default(System.DateTime?), System.DateTime? lastModified = default(System.DateTime?), RecommendedIndexTypes? indexType = default(RecommendedIndexTypes?), string schema = default(string), string table = default(string), System.Collections.Generic.IList<string> columns = default(System.Collections.Generic.IList<string>), System.Collections.Generic.IList<string> includedColumns = default(System.Collections.Generic.IList<string>), string indexScript = default(string), System.Collections.Generic.IList<OperationImpact> estimatedImpact = default(System.Collections.Generic.IList<OperationImpact>), System.Collections.Generic.IList<OperationImpact> reportedImpact = default(System.Collections.Generic.IList<OperationImpact>))
+            : base(location, name, id, type, tags)
+        {
+            Action = action;
+            State = state;
+            Created = created;
+            LastModified = lastModified;
+            IndexType = indexType;
+            Schema = schema;
+            Table = table;
+            Columns = columns;
+            IncludedColumns = includedColumns;
+            IndexScript = indexScript;
+            EstimatedImpact = estimatedImpact;
+            ReportedImpact = reportedImpact;
+        }
+
+        /// <summary>
+        /// Gets the proposed index action. We suggest user to create missing
+        /// index, drop unused index or to rebuild already existing index to
+        /// improve its performance. Possible values are 'Create', 'Drop',
+        /// 'Rebuild'. Possible values include: 'Create', 'Drop', 'Rebuild'
+        /// </summary>
```

Comment:
It would be good to reference RecommendedIndexActions class instead of writing out all the possible values here.